### PR TITLE
mgr/dashboard:minimize console log traces of Ceph backend API tests

### PR DESCRIPTION
**Pull request by aaSharma14**
Changed the log levels in some files to reduce the console logs when Ceph API tests are run

Fixes: https://tracker.ceph.com/issues/45822
Signed-off-by: Aashish Sharma <aasharma@redhat.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by aaSharma14** [comment]
Below are the log traces by running "test_mgr_module". In these logs the tests ran without an error.

[root@ceph dashboard]# run_teuthology_tests tasks.mgr.dashboard.test_mgr_module
Using guessed paths /ceph/build/lib/ ['/ceph/qa', '/ceph/src/pybind']
Using guessed paths /ceph/build/lib/ ['/ceph/qa', '/ceph/src/pybind']
2020-07-10 08:28:27,280.280 INFO:__main__:Starting test: test_get (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)
2020-07-10 08:28:29,167.167 INFO:__main__:test_get (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) ... ok
2020-07-10 08:28:29,168.168 INFO:__main__:Stopped test: test_get (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) in 1.887247s
2020-07-10 08:28:29,168.168 INFO:__main__:Starting test: test_list_disabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)
2020-07-10 08:28:32,240.240 INFO:__main__:test_list_disabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) ... ok
2020-07-10 08:28:32,240.240 INFO:__main__:Stopped test: test_list_disabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) in 3.071798s
2020-07-10 08:28:32,240.240 INFO:__main__:Starting test: test_list_enabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)
2020-07-10 08:28:40,876.876 INFO:__main__:test_list_enabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) ... ok
2020-07-10 08:28:40,877.877 INFO:__main__:Stopped test: test_list_enabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) in 8.636208s
2020-07-10 08:28:40,877.877 INFO:__main__:Starting test: test_put (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)
2020-07-10 08:28:44,910.910 INFO:__main__:test_put (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) ... ok
2020-07-10 08:28:44,911.911 INFO:__main__:Stopped test: test_put (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) in 4.033738s
2020-07-10 08:28:44,911.911 INFO:__main__:
2020-07-10 08:28:44,911.911 INFO:__main__:----------------------------------------------------------------------
2020-07-10 08:28:44,911.911 INFO:__main__:Ran 4 tests in 53.906s
2020-07-10 08:28:44,912.912 INFO:__main__:
2020-07-10 08:28:44,912.912 INFO:__main__:OK

**Comment by aaSharma14** [comment]
Below are the log traces of "test_mgr_module" with traceback..when there is a test failure.

[root@ceph dashboard]# run_teuthology_tests tasks.mgr.dashboard.test_mgr_module
Using guessed paths /ceph/build/lib/ ['/ceph/qa', '/ceph/src/pybind']
Using guessed paths /ceph/build/lib/ ['/ceph/qa', '/ceph/src/pybind']
2020-07-10 09:38:33,501.501 INFO:__main__:Starting test: test_get (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)
2020-07-10 09:38:35,330.330 INFO:__main__:test_get (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) ... ok
2020-07-10 09:38:35,330.330 INFO:__main__:Stopped test: test_get (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) in 1.829103s
2020-07-10 09:38:35,330.330 INFO:__main__:Starting test: test_list_disabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)
2020-07-10 09:38:38,431.431 INFO:__main__:test_list_disabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) ... ok
2020-07-10 09:38:38,431.431 INFO:__main__:Stopped test: test_list_disabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) in 3.10064s
2020-07-10 09:38:38,431.431 INFO:__main__:Starting test: test_list_enabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)
2020-07-10 09:38:51,987.987 INFO:__main__:test_list_enabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) ... ok
2020-07-10 09:38:51,987.987 INFO:__main__:Stopped test: test_list_enabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) in 13.555346s
2020-07-10 09:38:51,987.987 INFO:__main__:Starting test: test_put (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)
2020-07-10 09:38:56,010.010 INFO:__main__:test_put (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) ... FAIL
2020-07-10 09:38:56,010.010 INFO:__main__:Stopped test: test_put (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) in 4.022864s
2020-07-10 09:38:56,011.011 INFO:__main__:
2020-07-10 09:38:56,011.011 INFO:__main__:======================================================================
2020-07-10 09:38:56,011.011 INFO:__main__:FAIL: test_put (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)
2020-07-10 09:38:56,011.011 INFO:__main__:----------------------------------------------------------------------
2020-07-10 09:38:56,011.011 INFO:__main__:Traceback (most recent call last):
2020-07-10 09:38:56,011.011 INFO:__main__:  File "/ceph/qa/tasks/mgr/dashboard/test_mgr_module.py", line 140, in test_put
2020-07-10 09:38:56,011.011 INFO:__main__:    self.assertStatus(201)
2020-07-10 09:38:56,012.012 INFO:__main__:  File "/ceph/qa/tasks/mgr/dashboard/helper.py", line 376, in assertStatus
2020-07-10 09:38:56,012.012 INFO:__main__:    self.assertEqual(self._resp.status_code, status)
2020-07-10 09:38:56,012.012 INFO:__main__:AssertionError: 200 != 201
2020-07-10 09:38:56,012.012 INFO:__main__:
2020-07-10 09:38:56,012.012 INFO:__main__:----------------------------------------------------------------------
2020-07-10 09:38:56,012.012 INFO:__main__:Ran 4 tests in 58.916s
2020-07-10 09:38:56,012.012 INFO:__main__:
2020-07-10 09:38:56,013.013 INFO:__main__:FAILED (failures=1)
2020-07-10 09:38:56,013.013 INFO:__main__:
2020-07-10 09:38:56,013.013 INFO:__main__:======================================================================
2020-07-10 09:38:56,013.013 INFO:__main__:FAIL: test_put (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)
2020-07-10 09:38:56,013.013 INFO:__main__:----------------------------------------------------------------------
2020-07-10 09:38:56,013.013 INFO:__main__:Traceback (most recent call last):
2020-07-10 09:38:56,013.013 INFO:__main__:  File "/ceph/qa/tasks/mgr/dashboard/test_mgr_module.py", line 140, in test_put
2020-07-10 09:38:56,013.013 INFO:__main__:    self.assertStatus(201)
2020-07-10 09:38:56,013.013 INFO:__main__:  File "/ceph/qa/tasks/mgr/dashboard/helper.py", line 376, in assertStatus
2020-07-10 09:38:56,014.014 INFO:__main__:    self.assertEqual(self._resp.status_code, status)
2020-07-10 09:38:56,014.014 INFO:__main__:AssertionError: 200 != 201
2020-07-10 09:38:56,014.014 INFO:__main__:

File: qa/tasks/cephfs/filesystem.py [hunk 1]
```diff
@@ -509,13 +509,13 @@ def reach_max_mds(self):
...
                     target = targets[0]
-                    log.info("deactivating rank %d" % target['rank'])
+                    log.debug("deactivating rank %d" % target['rank'])
                     self.deactivate(target['rank'])
                     status = self.wait_for_daemons(skip_max_mds_check=True)
                     count = len(list(self.get_ranks(status=status)))
             except:
                 # In Mimic, deactivation is done automatically:
-                log.info("Error:\n{}".format(traceback.format_exc()))
+                log.debug("Error:\n{}".format(traceback.format_exc()))
```
[the remark this answers is not in the document]
Comment (aaSharma14) [reply qa/tasks/cephfs/filesystem.py:518]: Thanks Ernesto, Will modify this.

**Comment by aaSharma14** [comment]
> Thanks a lot, @aaSharma14 !
> 
> I'm curious: for the API tests, do we really need to modify the CephFS tests?
> 
> I'm not sure about this approach (reducing the priority of the logs messages), and most of them seem to fit well under the `info` level (reporting config changes & system operations). Could we instead increase the level of displayed traces (same effect but different meaning)?
> 
> Additionally, could you please display in the PR summary a brief example of the log traces before & after this change? Thanks!
> 
> Also added a couple of folks as reviewers, as I think their inputs are valuable for this improvement.

While running the API tests, CephFS tests also gets called a number of times from the vstart_runner.py (for example it shows the daemon health states and the MDS states while running the tests), so to make the test logs silent when everything runs fine..we need to change the log levels in that file too.

File: qa/tasks/cephfs/filesystem.py [hunk 2]
```diff
@@ -515,7 +515,7 @@ def reach_max_mds(self):
                     count = len(list(self.get_ranks(status=status)))
             except:
                 # In Mimic, deactivation is done automatically:
-                log.debug("Error:\n{}".format(traceback.format_exc()))
+                log.info("Error:\n{}".format(traceback.format_exc()))
```
Comment (tchaikov) [inline qa/tasks/cephfs/filesystem.py]: could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes

and "update" is far from informative, please summarize the change in the title, and put the rationales in the commit message.

**Comment by aaSharma14** [comment]
jenkins test dashboard

**Review by rishabh-d-dave — COMMENTED** [review]
LGTM. Waiting for the rebase.

**Review by epuertat — APPROVED** [review]
Thanks! Looks good to me, @aaSharma14. Let's see what others think (like @rishabh-d-dave or @batrick, as this impacts also their tests)...

**Review by batrick — APPROVED** [review]

**Comment by aaSharma14** [comment]
jenkins test dashboard

**Comment by aaSharma14** [comment]
Attached below are the API console logs after this improvement:-

[ApiTestLogs.txt](https://github.com/ceph/ceph/files/5733864/ApiTestLogs.txt)

**Comment by batrick** [comment]
cephfs tests pass

https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20201222.185105

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-12-30_06:47:01-rados-wip-kefu-testing-2020-12-30-1123-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-12-31_04:10:08-rados-wip-kefu-testing-2020-12-30-1123-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-12-31_09:50:24-rados-wip-kefu-testing-2020-12-31-1427-distro-basic-smithi/

test failures tracked by

- https://tracker.ceph.com/issues/46323

**Pull request merged**